src/Symfony/Component/Validator/Constraints/CssColor.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\InvalidArgumentException;
+
+/**
+ * @Annotation
+ * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
+ *
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class CssColor extends Constraint
+{
+    public const HEX_LONG = 'hex_long';
+    public const HEX_LONG_WITH_ALPHA = 'hex_long_with_alpha';
+    public const HEX_SHORT = 'hex_short';
+    public const HEX_SHORT_WITH_ALPHA = 'hex_short_with_alpha';
+    public const BASIC_NAMED_COLORS = 'basic_named_colors';
+    public const EXTENDED_NAMED_COLORS = 'extended_named_colors';
+    public const SYSTEM_COLORS = 'system_colors';
+    public const KEYWORDS = 'keywords';
+    public const RGB = 'rgb';
+    public const RGBA = 'rgba';
+    public const HSL = 'hsl';
+    public const HSLA = 'hsla';
+    public const INVALID_FORMAT_ERROR = '454ab47b-aacf-4059-8f26-184b2dc9d48d';
+
+    protected static $errorNames = [
+        self::INVALID_FORMAT_ERROR => 'INVALID_FORMAT_ERROR',
+    ];
+
+    /**
+     * @var string[]
+     */
+    private static $validationModes = [
+        self::HEX_LONG,
+        self::HEX_LONG_WITH_ALPHA,
+        self::HEX_SHORT,
+        self::HEX_SHORT_WITH_ALPHA,
+        self::BASIC_NAMED_COLORS,
+        self::EXTENDED_NAMED_COLORS,
+        self::SYSTEM_COLORS,
+        self::KEYWORDS,
+        self::RGB,
+        self::RGBA,
+        self::HSL,
+        self::HSLA,
+    ];
+
+    public $message = 'This value is not a valid hexadecimal color.';
```

Comment:
Is this message still accurate? After all, this constraint is able to validate more color formats than hex.